tools/rabbitmq-quorum-migration.sh

```diff
@@ -0,0 +1,56 @@
+#! /usr/bin/bash
+
+set -ex
+
+RABBITMQ_SERVICES_TO_RESTART=barbican,blazar,cinder,cloudkitty,designate,heat,ironic,keystone,magnum,manila,neutron,nova,octavia
+RABBITMQ_CONTAINER_NAME=rabbitmq
+
+if [[ ! $KAYOBE_CONFIG_PATH ]]; then
+    echo "Environment variable \$KAYOBE_CONFIG_PATH is not defined"
+    echo "Ensure your environment is set up to run kayobe commands"
+    exit 2
+fi
+
+if [[ ! "$1" = "--skip-checks" ]]; then
+    # Assert that clocks are synced
+    if ! kayobe overcloud host command run -l controllers -b --command "timedatectl status | grep 'synchronized: yes'"; then
+        echo "Failed precheck: Time not synced on controllers"
+        echo "Use 'timedatectl status' to check sync state"
+        echo "Either wait for sync or use 'chronyc makestep'"
+        exit 1
+    fi
+    kayobe overcloud service configuration generate --node-config-dir /tmp/rabbit-migration --kolla-tags none
+    # Assert that Quorum is enabled and HA is not
+    if ! grep 'om_enable_rabbitmq_quorum_queues: true' $KOLLA_CONFIG_PATH/globals.yml || grep 'om_enable_rabbitmq_high_availability: true' $KOLLA_CONFIG_PATH/globals.yml; then
+        echo "Failed precheck: om_enable_rabbitmq_quorum_queues must be enabled, om_enable_rabbitmq_high_availability must be disabled"
+        exit 1
+    fi
+fi
+
+# Generate new config, stop services using rabbit, and reset rabbit state
+kayobe overcloud service configuration generate --node-config-dir /etc/kolla --kolla-skip-tags rabbitmq-ha-precheck &&
+kayobe kolla ansible run "stop --yes-i-really-really-mean-it" -kt $RABBITMQ_SERVICES_TO_RESTART &&
+kayobe kolla ansible run rabbitmq-reset-state
+if [[ ! "$1" = "--skip-checks" ]]; then
+    # Assert that Queues and exchanges have stopped (excluding those with 'amq.' in the name)
+    if kayobe overcloud host command run -l controllers -b --command \
+    "(docker exec $RABBITMQ_CONTAINER_NAME rabbitmqctl list_queues name --silent && \
+      docker exec $RABBITMQ_CONTAINER_NAME rabbitmqctl list_exchanges name --silent) | \
+      grep -v '^$' | (! grep -v 'amq.')"; then
+        echo "Failed check: RabbitMQ has not stopped properly"
+        exit 1
+fi
+
+# Redeploy with Quorum Queues enabled
+kayobe kolla ansible run deploy -kt $RABBITMQ_SERVICES_TO_RESTART
```

Comment:
We already generated config, so can use `deploy-containers`?

Reply:
Done